### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by SwampertX**

### CODE REVIEW

File: src/closure.ts [hunk 1]
```diff
@@ -20,8 +20,8 @@ const closureToJS = (value: Closure, context: Context, klass: string) => {
   })
   Object.setPrototypeOf(DummyClass, () => undefined)
   Object.defineProperty(DummyClass, 'Inherits', {
-    value: (Parent: Value) => {
```
Comment (openorclose) [inline src/closure.ts]: This is for OOP. Doesn't make sense for it to be renamed to tail.

File: src/createContext.ts [hunk 2]
```diff
@@ -32,21 +32,21 @@ export const createEmptyContext = <T>(
   runtime: createEmptyRuntime()
 })
 
-export const ensureGlobalEnvironmentExist = (context: Context) => {
+export const ensureGlobalFrameExist = (context: Context) => {
```
Comment (openorclose) [inline src/createContext.ts]: Why is this being renamed to frame?

File: src/closure.ts (hunk 1 above)
Comment (martin-henz) [reply src/closure.ts]: I agree. @SwampertX please change this back to Parent.

File: src/createContext.ts (hunk 2 above)
Comment (martin-henz) [reply src/createContext.ts]: Yes, this should be ensureGlobalEnvironmentExist.

Another question: Why does a Context need multiple Environments?

**Pull request merged**